src/main/java/eu/mcdb/ban_announcer/PunishmentAction.java

```diff
@@ -48,6 +55,18 @@ public void setDuration(String duration) {
         this.duration = duration.equals("") ? "unknown" : duration;
     }
 
+    Date now = new Date();
+    SimpleDateFormat dateFormat = new SimpleDateFormat("MM-dd-yyyy");
+    SimpleDateFormat timeFormat = new SimpleDateFormat("hh:mm:ss aa");
+
+    public void setTime() {
+        this.time = timeFormat.format(now);
+    }
+
+    public void setDate() {
+        this.date = dateFormat.format(now);
+    }
+

```

Comment:
Typically a setter should accept one parameter, those method names are confusing, also they could be set in this class constructor.